pkg/controller/runtime/internal/qruntime/internal/containers/priority_queue.go

```diff
@@ -26,7 +26,7 @@ type PriorityQueue[T comparable] struct {
 // Push item to the queue with releaseAfter time.
 //
 // If the item is not in the queue, it will be added.
-// If the item is in the queue, and releaseAfter is greater than the existing releaseAfter, it will be re-added in the new position.
+// If the item is in the queue, and releaseAfter is greater than the existing releaseAfter, it will be not be added to the queue.
```

Comment:
this comment looks to be off now, as it now re-inserts with smaller timestamp, and tests should be adjusted

Reply:
and we need tests for new behavior, the current thing doesn't look good

Reply:
Updated